atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -75,22 +93,58 @@ public void batch_mutate(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_m
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
         checkLimitAndCall(() -> {
             super.batch_mutate(mutation_map, consistency_level);
+
+            qosMetrics.updateWriteCount();
+            qosMetrics.updateBytesWritten(getApproximateWriteByteCount(mutation_map));
             return null;
         });
     }
 
+    private int getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map) {
+        int approxBytesForKeys = mutation_map.keySet().stream().mapToInt(e -> e.array().length).sum();
+        int approxBytesForValues = mutation_map.values().stream()
+                .mapToInt(singleMap -> {
+                    int approximateBytesInStrings = singleMap.keySet().stream().mapToInt(String::length).sum();
+                    int approximateBytesInMutations = singleMap.values().stream()
+                            .mapToInt(listOfMutations -> listOfMutations.stream()
+                                    .mapToInt(mutation -> SerializationUtils.serialize(mutation).length)
+                                    .sum())
+                            .sum();
+                    return approximateBytesInStrings + approximateBytesInMutations;
+                })
+                .sum();
+        return approxBytesForKeys + approxBytesForValues;
+    }
+
     @Override
     public CqlResult execute_cql3_query(ByteBuffer query, Compression compression, ConsistencyLevel consistency)
             throws InvalidRequestException, UnavailableException, TimedOutException, SchemaDisagreementException,
             TException {
+
+        // This could be used for updating the schema mutation lock,
+        //  "UPDATE "%s" SET value = %s WHERE key = %s AND column1 = %s AND column2 = -1;"
+
+        // or to read the timestamp bound
+        // SELECT %s, %s FROM %s WHERE key=%s;
+
+        // Or for sweep?
+        // Should we consider all of these reads when recording metrics?
```

Comment:
I think we should examine the `CqlResult` object returned. We can probably use the size of the rows within this object.

Reply:
Done.